auto_submit/cloudbuild_auto_submit.yaml

```diff
@@ -1,20 +1,32 @@
 # Provide instructions for google Cloud Build to auto-build flutter
 # auto-submit bot to flutter-dashboard project. Auto-build will be triggered
-# by daily schedule on `main` branch. This cloudbuild calls two cloudbuild
-# configurations.
+# by daily schedule on `main` branch. This cloudbuild calls an additional
+# cloudbuild configuration responsible for deployment.
 #
-# The first job is for generating the docker image with build provenance,
-# and the second uses the generated docker image and deploys it to app engine.
+# This job is for generating the docker image with build provenance,
+# and the deployment job uses the generated docker image and deploys it to
+# App Engine.
 
 steps:
+  # Build docker image
+  - name: 'us-docker.pkg.dev/cloud-builders/ga/v1/docker'
+    args: ['build', '-t', 'us-docker.pkg.dev/$PROJECT_ID/appengine/auto-submit.version-$SHORT_SHA', 'auto_submit']
+
+  # Trigger the cloud build that deploys the docker image
   - name: gcr.io/cloud-builders/gcloud
     entrypoint: '/bin/bash'
     args:
       - '-c'
       - |-
-        gcloud builds submit --config auto_submit/cloudbuild_auto_submit_docker.yaml \
-          --substitutions="SHORT_SHA=$SHORT_SHA"
-        gcloud builds submit --config auto_submit/cloudbuild_auto_submit_deploy.yaml \
-          --substitutions="_GAE_PROMOTE=$_GAE_PROMOTE,SHORT_SHA=$SHORT_SHA"
+        gcloud builds submit \
+          --config auto_submit/cloudbuild_auto_submit_deploy.yaml \
+          --substitutions="_GAE_PROMOTE=$_GAE_PROMOTE,SHORT_SHA=$SHORT_SHA" \
+          --async
 
 timeout: 1200s
+
+images: ['us-docker.pkg.dev/$PROJECT_ID/appengine/auto-submit.version-$SHORT_SHA']
+
+# If build provenance is not generated, the docker deployment will fail.
+options:
```

Comment:
Should we have a test for this change to prevent regressions? We could update the [readiness check request](https://cs.opensource.google/flutter/cocoon/+/main:app_dart/lib/src/request_handlers/readiness_check.dart) to verify the provenance exists.

Reply:
When is the readiness check endpoint currently called?

Reply:
Oh, [I think I answered my own question](https://cs.opensource.google/flutter/cocoon/+/main:app_dart/app.yaml;l=9), lol

I would think testing here would be too late, as the artifact would already be deployed. The intention of this step is to break the build of an artifact without provenance before it gets a chance to be deployed at all.

Reply:
The readiness check indicates whether AppEngine can serve traffic to the current instance. While it is late, it will block deployments that violate the SLSA3 compliance.